tensorflow/lite/delegates/nnapi/nnapi_delegate.cc

```diff
@@ -655,7 +655,23 @@ NNMemory::NNMemory(const NnApi* nnapi, const char* name, size_t size) {
   if (name && size > 0) {
     nnapi_ = nnapi;
     byte_size_ = size;
+#ifdef __ANDROID__
     fd_ = nnapi_->ASharedMemory_create(name, size);
+#else
+    // Each call to ASharedMemory_create produces a unique memory space, hence
+    // name should not be used to create the shared memory file, otherwise
+    // two calls to create memory regions using the same 'name', will collide.
+    char shm_name_buffer[L_tmpnam];
+    if (tmpnam(shm_name_buffer) == nullptr) {
+      throw new std::runtime_error("NN Memory failed");
+    }
+    // tmpnam will produce a string containing with slashes, but shm_open
+    // won't like that.
+    shm_region_name = std::string(name) + std::string(shm_name_buffer);
```

Comment:
I guess it's fine to ignore the "name".

Reply:
It has no impact on the functionality, howerver, if the name was not be used, it would be not evident what shared memory object are created by NNMemory::NNMemory(). 
Do you insists on ignoring the name?

Reply:
ack, fine to keep it as-is